edx_toggles/toggles/internal/waffle/legacy.py

```diff
@@ -0,0 +1,214 @@
+"""
+This module contains legacy code for backward compatibility. Waffle flag and switch objects previously required the
+creation of namespace objects. The namespace features were all moved to the WaffleSwitch/Flag classes.
+
+To upgrade your code, use the following guidelines. Where previously you had::
+
+    SOME_NAMESPACE = WaffleSwitchNamespace("some_namespace")
+    SOME_SWITCH = WaffleSwitch(SOME_NAMESPACE, "some_switch", module_name=__name__)
+
+You should now write::
+
+    SOME_SWITCH = WaffleSwitch("some_namespace.some_switch", module_name=__name__)
+
+And similarly for waffle flags, replace::
+
+    SOME_NAMESPACE = WaffleFlagNamespace("some_namespace", log_prefix="some_namespace")
+    SOME_FLAG = WaffleFlag(SOME_NAMESPACE, "some_flag", module_name=__name__)
+
+by::
+
+    SOME_FLAG = WaffleFlag("some_namespace.some_flag", module_name=__name__, log_prefix="some_namespace")
```

Comment:
I added the imports to the docstring. I think it makes more sense to keep this here in the legacy module, because this is where people will look first after they upgrade to 1.2.0.